packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -38,6 +37,34 @@ class SharedPreferences {
     return _completer.future;
   }
 
+  static SharedPreferencesPlatform _platform;
+
+  /// Sets a custom [SharedPreferencesPlatform].
+  ///
+  /// This property can be set to use a custom platform implementation.
+  static set platform(SharedPreferencesPlatform platform) {
+    assert(_platform == null);
+    _platform = platform;
+  }
+
+  /// The SharedPreferences platform that's used by the plugin.
+  ///
+  /// The default value is [MethodChannelSharedPreferences] on Android or iOS.
+  static SharedPreferencesPlatform get platform {
+    if (_platform == null) {
+      switch (defaultTargetPlatform) {
```

Comment:
I don't think the target platform is actually what we should be using in these switches. The implementation that's built into the app is going to be a function of the host build, not the target.

For instance, imagine someone building on Windows, but wanting to get a sort of "iOS Simulator lite" experience for testing, sets defaultTargetPlatform to iOS to get iOS look and feel. Having that affect the backend it tries to load here doesn't seem like it makes sense. (Yes, it could be overridden, and if/when the Dart generation is added it presumably would be automatically, but in that case why have this code at all?)